core/metadata.cpp

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "metadata.h"
+#include "exif.h"
+#include "qthelper.h"
+#include <QString>
+#include <QFile>
+#include <QDateTime>
+
+// Fetch quint16 in big endian mode from QFile and return 0 on error.
+// This is a very specialized function for parsing JPEGs, therefore we can get away with such an in-band error code.
+static inline quint16 getShortBE(QFile &f)
+{
+	unsigned char buf[2];
+	if (f.read(reinterpret_cast<char *>(buf), 2) != 2)
```

Comment:
what's a reinterpret_cast ?
(I realize that this is just copied code and I should have asked when this first came in... but still)

Reply:
`reinterpret_cast<char *>(buf)` is equivalent to `(char*)buf`, but telling the compiler that we know that we're doing something potentially dangerous.